### PR TITLE
Even odd checker

### PR DESCRIPTION
**Pull request by robiel0143**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations
- [ ] The function is not called in the function file

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] There are no commented lines of code
- [ ] There are no `print` statements anywhere
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/even_odd_checker.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Module contents:
+    - even_odd_checker: checks if a given number is even or odd.
+
+Created on 2024-12-31
+Author: Robel Mengsteab
+"""
+
+
+def even_odd_checker(n: int) -> str:
```
Comment (Khusro-S) [inline solutions/even_odd_checker.py:14]: Even though `n` is clear in this context, it’s best practice to use more descriptive names. Something like `number` or a name that shows what it represents would make the code easier to understand.

File: solutions/tests/test_even_odd_checker.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        """It should return even"""
+        self.assertEqual(even_odd_checker(-23469574390), "Even")
+
+    def test_positive_whole_float(self):
+        """It should return Even"""
+        self.assertEqual(even_odd_checker(6.0), "Even")
+
+    def test_negative_whole_float(self):
+        """It should return Odd"""
+        self.assertEqual(even_odd_checker(-99), "Odd")
```
Comment (Khusro-S) [inline solutions/tests/test_even_odd_checker.py:76]: If I'm not mistaken, I think you meant (even_odd_checker(-99.0), "Odd") here, since it's testing a negative whole float?

**Review by Khusro-S — COMMENTED** [review]
This was very straightforward and easy to understand, great job! I just pointed out a few of very minor suggestions.

File: solutions/even_odd_checker.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    """Checks if a number is even or odd.
+
+    Determines if the input number is even or odd and validates the input.
+
+    Parameters:
+        n: int, the number to check.
```
Comment (Khusro-S) [inline solutions/even_odd_checker.py:20]: Same goes here, the variable name as I mentioned above.
